x-pack/plugins/lens/public/xy_visualization/expression_thresholds.tsx

```diff
@@ -0,0 +1,191 @@
+                details: formatter.convert(row[yConfig.forAccessor]),
+              }))}
+              domainType={
+                yConfig.axisMode === 'bottom'
+                  ? AnnotationDomainType.XDomain
+                  : AnnotationDomainType.YDomain
+              }
+              style={{
+                line: {
+                  ...sharedStyle,
+                  opacity: 1,
+                },
+              }}
+            />
+          );
+
+          if (yConfig.fill && yConfig.fill !== 'none') {
+            const isFillAbove = yConfig.fill === 'above';
+            const indexFromSameType = groupedByDirection[yConfig.fill].findIndex(
+              ({ forAccessor }) => forAccessor === yConfig.forAccessor
+            );
+            const shouldCheckNextThreshold =
+              indexFromSameType < groupedByDirection[yConfig.fill].length - 1;
+            annotations.push(
+              <RectAnnotation
+                {...props}
+                id={`${layerId}-${yConfig.forAccessor}-rect`}
+                key={`${layerId}-${yConfig.forAccessor}-rect`}
+                dataValues={table.rows.map(() => {
+                  if (yConfig.axisMode === 'bottom') {
+                    return {
+                      coordinates: {
+                        x0: isFillAbove ? row[yConfig.forAccessor] : undefined,
+                        y0: undefined,
+                        x1: isFillAbove
+                          ? shouldCheckNextThreshold
+                            ? row[
+                                groupedByDirection[yConfig.fill!][indexFromSameType + 1].forAccessor
+                              ]
+                            : undefined
+                          : row[yConfig.forAccessor],
+                        y1: undefined,
+                      },
+                      header: columnToLabelMap[yConfig.forAccessor],
+                      details: formatter.convert(row[yConfig.forAccessor]),
+                    };
+                  }
+                  return {
+                    coordinates: {
+                      x0: undefined,
+                      y0: isFillAbove ? row[yConfig.forAccessor] : undefined,
+                      x1: undefined,
+                      y1: isFillAbove
+                        ? shouldCheckNextThreshold
+                          ? row[
+                              groupedByDirection[yConfig.fill!][indexFromSameType + 1].forAccessor
+                            ]
+                          : undefined
+                        : row[yConfig.forAccessor],
+                    },
+                    header: columnToLabelMap[yConfig.forAccessor],
+                    details: formatter.convert(row[yConfig.forAccessor]),
+                  };
+                })}
+                style={{
+                  ...sharedStyle,
+                  fill: (yConfig.color || defaultColor) ?? '#f00',
+                  opacity: 0.3,
```

Comment:
As mentioned in the design notes, I recommend that threshold fills be 10% opacity.

```suggestion
                  opacity: 0.1,
```